pkg/controllers/node/kdd.go

```diff
@@ -0,0 +1,199 @@
+package node
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	bapi "github.com/projectcalico/libcalico-go/lib/backend/api"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	cerrors "github.com/projectcalico/libcalico-go/lib/errors"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type allocation struct {
+	handle     string
+	attributes map[string]string
+}
+
+// syncDeleteKDD cleans up any IPAM resources which should no longer exist based on nodes in the cluster.
+// It returns an error if it is determined that there are resources which should be cleaned up, but is unable to do so.
+// It does not return an error if it is successful, or if there is no action to take.
+func (c *NodeController) syncDeleteKDD() error {
+	// Get the backend client.
+	type accessor interface {
+		Backend() bapi.Client
+	}
+	bc := c.calicoClient.(accessor).Backend()
```

Comment:
This is a strange pattern...Why does it have to be this way? Why not just expose `Backend()` in the public interface?

Reply:
Guess we use it enough now. It's more a historical thing - nothing was ever supposed to need to use the backend client like this when the interface was originally defined, but now we've found it useful. 

Suppose we could just add it to the public interface.